test/features/webapp-prompt/webapp-prompt.test.js

```diff
@@ -0,0 +1,153 @@
+import { expect } from '@esm-bundle/chai';
+import sinon, { stub } from 'sinon';
+import pepPromptContent from './mocks/pep-prompt-content.js';
+
+describe('PEP', () => {
+  let clock;
+  let allSelectors;
+  let defaultConfig;
+  let mockRes;
+
+  beforeEach(async () => {
+    clock = sinon.useFakeTimers({
+      toFake: ['setTimeout'],
+      shouldAdvanceTime: true,
+    });
+    // We need to import the utilities after mocking setTimeout to ensure
+    // their setTimeout calls use Sinon's mocked implementation.
+    // Importing before mocking would lead to a 5s PEP timeout, exceeding the 2s test limit.
+    const { allSelectors: importedAllSelectors, defaultConfig: importedDefaultConfig, mockRes: importedMockRes } = await import('./test-utilities.js');
+    allSelectors = importedAllSelectors;
+    defaultConfig = importedDefaultConfig;
+    mockRes = importedMockRes;
+
+    document.body.innerHTML = `<div class="${allSelectors.fedsUtilities.replace('.', '')}" style="position:relative">
+    <div id="${allSelectors.appSwitcher.replace('#', '')}" tabindex="0">App Switcher</div>
+    </div>`;
+    stub(window, 'fetch').callsFake(async (url) => {
+      if (url.includes('pep-prompt-content.plain.html')) return mockRes({ payload: pepPromptContent({ ...defaultConfig }) });
+      return null;
+    });
+  });
+
+  afterEach(() => {
+    sinon.restore();
+    clock.restore();
+    document.body.innerHTML = '';
+    document.cookie = `${document.cookie};expires=Thu, 01 Jan 1970 00:00:00 GMT;path=/`;
+  });
+
+  describe('PEP rendering tests', () => {
+    it('should render PEP', async () => {
+      await (await import('./test-utilities.js')).initPep({});
```

Comment:
Is this imported here to avoid similar issues to the mocked sinon timeout // in-module caches?

Reply:
Yes, it's imported there to avoid mocked timeout issues. Now that I look at it, we don't need to import it again and again for every test. It can be stored in a variable. Thanks for drawing attention to it 👍